### PR TITLE
ci(Mergify): configuration update

### PR DESCRIPTION
**Pull request by Phoenix500526**
Depends-On: #776 
This change has been made by @Phoenix500526 from Mergify config editor.

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -1,15 +1,67 @@
...
-      - status-success="codecov/patch"
-      - status-success="codecov/project"
+      - or:
+        - "#approved-reviews-by >= 2"
+        - and:
+          - author = dependabot[bot]
+          - "#approved-reviews-by>=1"
+      - or:
+        - "#check-failure=0"
+        - and:
```
Comment (rogercloud) [inline .mergify.yml:11]: code coverage check should be restricted.

File: .mergify.yml [hunk 2]
```diff
@@ -1,15 +1,67 @@
...
+
+  - name: automatic approval for Dependabot pull requests
+    conditions:
+      - author=dependabot[bot]
+    actions:
+      review:
+        type: APPROVE
+        message: Automatically approving dependabot
+
+  - name: comment when a pull request is merged
```
Comment (rogercloud) [inline .mergify.yml:45]: "when a pull request cannot be merged"

Comment (Phoenix500526) [reply .mergify.yml:45]: Done

**Review by rogercloud — APPROVED** [review]

File: .mergify.yml [hunk 3]
```diff
@@ -1,15 +1,55 @@
...
+      - and:
+          - "#check-failure>0"
+          - -check-failure~=codecov/.*
+    actions:
+      comment:
+        message: "@{{author}} Convert your pr to draft since CI failed"
+      edit:
+        draft: true
+
+  - name: automatic approval for Dependabot pull requests
```
Comment (bsbds) [inline .mergify.yml:25]: Will this cause failed depbot PR be merged automatically?

File: .mergify.yml [hunk 4]
```diff
@@ -1,15 +1,55 @@
...
+    conditions:
+      - author=dependabot[bot]
+    actions:
+      review:
+        type: APPROVE
+        message: Automatically approving dependabot
+
+  - name: comment when a pull request cannot be merged
```
Comment (bsbds) [inline .mergify.yml:33]: This comment action seems redundant since it's the default behaviour.

File: .mergify.yml (hunk 3 above)
Comment (Phoenix500526) [reply .mergify.yml:25]: It won't since a failed depbot PR doesn't  meet the "Automatic merge on approval" rule

**Review by bsbds — APPROVED** [review]

**Pull request merged**